### PR TITLE
DDF-1192: Added OWSP exclusions for false positives, and updated restlet to version 2.1.4, shiro to version 1.2.3 and Xalan to version 2.7.2 to address other high severity vulnerabilities.

### PR DESCRIPTION
**Pull request by lessarderic**
[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-platform/49)

### CODE REVIEW

**Comment by lessarderic** [comment]
@shaundmorris, @pklinef

**Comment by shaundmorris** [comment]
Review status: all files reviewed, 2 unresolved discussions, some commit checks pending.
Reviewed files:
- compression/platform-compression-exi/dependency-check-maven-config.xml @ r1
- compression/platform-compression-exi/pom.xml @ r1
- error/platform-error-impl/pom.xml @ r1
- persistence/platform-persistence-core-impl/pom.xml @ r1
- pom.xml @ r1
- security/filter/security-filter-login/pom.xml @ r1
- security/handler/pom.xml @ r1
- security/pom.xml @ r1
- solr/platform-solr-server-standalone/dependency-check-maven-config.xml @ r1
- solr/platform-solr-server-standalone/pom.xml @ r1
- solr/pom.xml @ r1

---

<sup>**[compression/platform-compression-exi/pom.xml, line 67 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/49#-JplonJkrzB25pG21rnk)** ([raw file](https://github.com/codice/ddf-platform/blob/680c0603055736af35965a697515c6dcb2e11030/compression/platform-compression-exi/pom.xml#L67)):</sup>
Okay so this was just an unused dependency in testing...

---

<sup>**[security/pom.xml, line 33 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/49#-JplpYksfKDcqctDL4bT)** ([raw file](https://github.com/codice/ddf-platform/blob/a6b9a6fa603a3f74357189967911f35f98a43ccc/security/pom.xml#L33)):</sup>
Since the parent is ddf-platform, couldn't we just use the shiro.version property defined in ddf-platform/pom.xml ?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/49)

**Comment by lessarderic** [comment]
Review status: all files reviewed, 2 unresolved discussions, some commit checks broke.

---

<sup>**[compression/platform-compression-exi/pom.xml, line 67 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/49#-JplonJkrzB25pG21rnk)** ([raw file](https://github.com/codice/ddf-platform/blob/680c0603055736af35965a697515c6dcb2e11030/compression/platform-compression-exi/pom.xml#L67)):</sup>
Correct

---

<sup>**[security/pom.xml, line 33 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/49#-JplpYksfKDcqctDL4bT)** ([raw file](https://github.com/codice/ddf-platform/blob/a6b9a6fa603a3f74357189967911f35f98a43ccc/security/pom.xml#L33)):</sup>
Good catch! Will fix.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/49)

**Comment by shaundmorris** [comment]
Review status: all files reviewed, 2 unresolved discussions, some commit checks broke.
Reviewed files:
- security/pom.xml @ r2

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/49)

File: security/pom.xml [hunk 1]
```diff
@@ -30,8 +30,6 @@
         <ddf.platform.app.version>2.7.0-SNAPSHOT</ddf.platform.app.version>
         <jasypt.version>1.9.0</jasypt.version>
         <wss4j.version>2.0.3</wss4j.version>
-        <shiro.version>1.2.1</shiro.version>
-        <karaf.version>2.4.1</karaf.version>
```
Comment (shaundmorris) [inline security/pom.xml]: good catch might as well do the karaf.version as well

**Comment by pklinef** [comment]
Can you create tickets for most of these dependencies updates with CVE information included?  We need a way to track that all of this needs to be fixed instead of comments.

**Comment by pklinef** [comment]
:+1:

**Comment by shaundmorris** [comment]
Completed final build check.

**Pull request merged**